src/auth/googleauth.ts

```diff
@@ -191,9 +198,32 @@ export class GoogleAuth<T extends AuthClient = JSONClient> {
     }
   }
 
-  private getProjectIdAsync(): Promise<string | null> {
+  /**
+   * A temporary method for internal `getProjectId` usages where `null` is
+   * acceptable. In a future major release, `getProjectId` should return `null`
+   * (as the `Promise<string | null>` base signature describes) and this private
+   * method should be removed.
+   *
+   * @returns Promise that resolves with project id (or `null`)
+   */
+  async #getProjectIdOptional(): Promise<string | null> {
+    try {
+      return await this.getProjectId();
+    } catch (e) {
+      if (
+        e instanceof Error &&
+        e.message === GoogleAuthExceptionMessages.NO_PROJECT_ID_FOUND
+      ) {
+        return null;
+      } else {
+        throw e;
+      }
+    }
+  }
```

Comment:
Leveraging the Promises API (similar to the previous PR) 😄 might simplify it a bit here again:

```suggestion
  #getProjectIdOptional(): Promise<string | null> {
    return this.getProjectId()
      .catch (e => {
        if (e.message === GoogleAuthExceptionMessages.NO_PROJECT_ID_FOUND) {
          return null;
        } else {
          throw e;
        }
      })
  }
```